### PR TITLE
Added comments for system tables

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
Added comments for all the system tables which are now shown in `system.tables` table.

```
SELECT
    concat(database, '.', name),
    comment
FROM system.tables
WHERE database = 'system'
LIMIT 10

Query id: 4144450d-9338-4539-98e0-e5510d8c99be

┌─concat(database, '.', name)───────────┬─comment────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ system.aggregate_function_combinators │ Contains a list of all available aggregate function combinators, which could be applied to aggregate functions and change the way they work.                                                               │
│ system.asynchronous_inserts           │ Contains information about pending asynchronous inserts in queue in server's memory.                                                                                                                       │
│ system.asynchronous_loader            │ Contains information and status for recent asynchronous jobs (e.g. for tables loading). The table contains a row for every job.                                                                            │
│ system.asynchronous_metric_log        │ Contains the historical values for system.asynchronous_metrics, which are saved once per minute.                                                                                                           │
│ system.asynchronous_metric_log_0      │                                                                                                                                                                                                            │
│ system.asynchronous_metrics           │ Contains metrics that are calculated periodically in the background. For example, the amount of RAM in use.                                                                                                │
│ system.backups                        │ Contains a list of all BACKUP or RESTORE operations with their current states and other propertis. Note, that table is not persistent and it shows only operations executed after the last server restart. │
│ system.build_options                  │ Contains a list of all build flags, compiler options and commit hash for used build.                                                                                                                       │
│ system.certificates                   │ Contains information about available certificates and their sources.                                                                                                                                       │
│ system.clusters                       │ Contains information about clusters defined in the configuration file or generated by a Replicated database.                                                                                               │
└───────────────────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added comments for all system tables which shown in `system.tables`.

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
Will continue later.

File: src/Storages/System/attachSystemTables.cpp [hunk 1]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemContributors>(context, system_database, "contributors", "Contains a list of all ClickHouse contributors <3");
+    attach<StorageSystemUsers>(context, system_database, "users", "Contains a list of all users accounts either configured at the server through configuration file or created via SQL.");
+    attach<StorageSystemRoles>(context, system_database, "roles", "Contains a list of all roles created at the server.");
+    attach<StorageSystemGrants>(context, system_database, "grants", "Contains the information about privileges granted to ClickHouse user accounts.");
+    attach<StorageSystemRoleGrants>(context, system_database, "role_grants", "Contains the role grants for users and roles. To add entries to this table, use `GRANT role TO user`. Using this table you may find out which roles are assigned to which users or which roles does user has.");
```
Comment (novikd) [inline src/Storages/System/attachSystemTables.cpp:125]:
```suggestion
    attach<StorageSystemRoleGrants>(context, system_database, "role_grants", "Contains the role grants for users and roles. To add entries to this table, use `GRANT role TO user`. Using this table you may find out which roles are assigned to which users or which roles a user has.");
```

File: src/Storages/System/attachSystemTables.cpp [hunk 2]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemCurrentRoles>(context, system_database, "current_roles", "Contains active roles of a current user. SET ROLE changes the contents of this table.");
+    attach<StorageSystemEnabledRoles>(context, system_database, "enabled_roles", "Contains all active roles at the moment, including current role of the current user and granted roles for current role.");
```
Comment (novikd) [inline src/Storages/System/attachSystemTables.cpp:127]: Can there be something more that `current role of the current user and granted roles for current role`?

File: src/Storages/System/attachSystemTables.cpp [hunk 3]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemDataTypeFamilies>(context, system_database, "data_type_families", "Contains a list of all available native data types along with all the aliases which a useful for compatibility with other databases.");
+    attach<StorageSystemCollations>(context, system_database, "collations", "Contains a list of all available collations for alphabetical comparison of strings.");
+    attach<StorageSystemTableEngines>(context, system_database, "table_engines", "Contains a list of all available table engines along with several flags whether a particular table engine supports some feature (e.g. settings, skipping indices, projections, replication, ttl, deduplication, parallel insert etc.)");
+    attach<StorageSystemContributors>(context, system_database, "contributors", "Contains a list of all ClickHouse contributors <3");
+    attach<StorageSystemUsers>(context, system_database, "users", "Contains a list of all users accounts either configured at the server through configuration file or created via SQL.");
```
[the remark this answers is not in the document]
Comment (novikd) [reply src/Storages/System/attachSystemTables.cpp:122]: or just leave users

File: src/Storages/System/attachSystemTables.cpp [hunk 4]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemRoleGrants>(context, system_database, "role_grants", "Contains the role grants for users and roles. To add entries to this table, use `GRANT role TO user`. Using this table you may find out which roles are assigned to which users or which roles does user has.");
+    attach<StorageSystemCurrentRoles>(context, system_database, "current_roles", "Contains active roles of a current user. SET ROLE changes the contents of this table.");
+    attach<StorageSystemEnabledRoles>(context, system_database, "enabled_roles", "Contains all active roles at the moment, including current role of the current user and granted roles for current role.");
+    attach<StorageSystemSettingsProfiles>(context, system_database, "settings_profiles", "Contains properties of configured setting profiles.");
+    attach<StorageSystemSettingsProfileElements>(context, system_database, "settings_profile_elements", "Describes the content of each settings profile configured on the server. Including settings contraints, roles and users that the setting applies to and parent settings profiles.");
```
Comment (novikd) [inline src/Storages/System/attachSystemTables.cpp:129]:
```suggestion
    attach<StorageSystemSettingsProfileElements>(context, system_database, "settings_profile_elements", "Describes the content of each settings profile configured on the server. Including settings constraints, roles and users for which the settings are applied, and parent settings profiles.");
```

File: src/Storages/System/attachSystemTables.cpp [hunk 5]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemRowPolicies>(context, system_database, "row_policies", "Contains filters for one particular table, as well as a list of roles and/or users which should use this row policy.");
+    attach<StorageSystemQuotas>(context, system_database, "quotas", "Contains information about quotas - a way how to limit resource usage over a period of time or track the use of resources.");
```
Comment (novikd) [inline src/Storages/System/attachSystemTables.cpp:131]: This looks quite strange. It explains not the information in the table, but the usage of quotas.

File: src/Storages/System/attachSystemTables.cpp [hunk 6]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemQuotaLimits>(context, system_database, "quota_limits", "Contains information about maximums for all intervals of all quotas. Any number of rows or zero can correspond to one quota.");
```
Comment (novikd) [inline src/Storages/System/attachSystemTables.cpp:132]: > Any number of rows or zero

Better to describe when info for the quota can be missing.

Comment (novikd) [reply src/Storages/System/attachSystemTables.cpp:132]:
```suggestion
    attach<StorageSystemQuotaLimits>(context, system_database, "quota_limits", "Contains information about maximums for all intervals of all quotas. Any number of rows or zero can correspond to specific quota.");
```

File: src/Storages/System/attachSystemTables.cpp [hunk 7]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemQuotaUsage>(context, system_database, "quota_usage", "Contains quota usage by the current user: how much is used and how much is left.");
+    attach<StorageSystemQuotasUsage>(context, system_database, "quotas_usage", "Contains quota usage by all users.");
+    attach<StorageSystemUserDirectories>(context, system_database, "user_directories", "Contains the information about configured user directories - directories on the file system from which ClickHouse server is allowed to read user provided data.");
+    attach<StorageSystemPrivileges>(context, system_database, "privileges", "Contains a list of all available privileges that could be granted to a user or role.");
+    attach<StorageSystemErrors>(context, system_database, "errors", "Contains a list of all error which have ever happened including the error code, last time and message with unsymbolized stacktrace.");
```
Comment (novikd) [inline src/Storages/System/attachSystemTables.cpp:137]:
```suggestion
    attach<StorageSystemErrors>(context, system_database, "errors", "Contains a list of all errors which have ever happened including the error code, last time and message with unsymbolized stacktrace.");
```

File: src/Storages/System/attachSystemTables.cpp [hunk 8]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemWarnings>(context, system_database, "warnings", "Contains warnings about server configuration to be displayed in clickhouse-client one any is connected.");
```
Comment (novikd) [inline src/Storages/System/attachSystemTables.cpp:138]:
```suggestion
    attach<StorageSystemWarnings>(context, system_database, "warnings", "Contains warnings about server configuration to be displayed by clickhouse-client right after it connects to the server.");
```

File: src/Storages/System/attachSystemTables.cpp [hunk 9]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemDataSkippingIndices>(context, system_database, "data_skipping_indices", "Contains all the information about all the data skipping indices in tables, similar to system.columns.");
+    attach<StorageSystemLicenses>(context, system_database, "licenses", "Сontains licenses of third-party libraries that are located in the contrib directory of ClickHouse sources.");
```
Comment (novikd) [inline src/Storages/System/attachSystemTables.cpp:140]: Does is contain info about `rust` folder?

File: src/Storages/System/attachSystemTables.cpp [hunk 10]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
-    attach<StorageSystemPrivileges>(context, system_database, "privileges");
-    attach<StorageSystemErrors>(context, system_database, "errors");
-    attach<StorageSystemWarnings>(context, system_database, "warnings");
-    attach<StorageSystemDataSkippingIndices>(context, system_database, "data_skipping_indices");
-    attach<StorageSystemLicenses>(context, system_database, "licenses");
-    attach<StorageSystemTimeZones>(context, system_database, "time_zones");
-    attach<StorageSystemBackups>(context, system_database, "backups");
-    attach<StorageSystemSchemaInferenceCache>(context, system_database, "schema_inference_cache");
-    attach<StorageSystemDroppedTables>(context, system_database, "dropped_tables");
+    attach<StorageSystemOne>(context, system_database, "one", "Used when the table is not specified explicitly, for example in queries like `SELECT 1`. Analog of the DUAL table in Oracle and MySQL.");
```
Comment (rschu1ze) [inline src/Storages/System/attachSystemTables.cpp:99]: Comment does not say what the system.one actually does. Could include "This table contains a single row with a single dummy UInt8 column containing the value 0."

Comment (rschu1ze) [reply src/Storages/System/attachSystemTables.cpp:99]: (I'd also omit the reference to DUAL tables. This is interesting but belongs IMHO more into the official docs.)

File: src/Storages/System/attachSystemTables.cpp [hunk 11]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemNumbers>(context, system_database, "numbers", "Generates all natural numbers, starting from 0 (to 2^64 - 1, and then again) in sorted order.", false);
+    attach<StorageSystemNumbers>(context, system_database, "numbers_mt", "Multithreaded version of `system.numbers`. The result could be out of order.", true);
+    attach<StorageSystemZeros>(context, system_database, "zeros", "Generated unlimited amount of non-materialized zeros.", false);
```
[the remark this answers is not in the document]
Comment (rschu1ze) [reply src/Storages/System/attachSystemTables.cpp:102]: I don't find `system.zeros`/`system.zeros_mt` documented on https://clickhouse.com/docs/en/operations/system-tables

File: src/Storages/System/attachSystemTables.cpp [hunk 12]
```diff
@@ -96,105 +96,105 @@ namespace DB
...
+    attach<StorageSystemZeros>(context, system_database, "zeros_mt", "Multithreaded version of system.zeros.", true);
+    attach<StorageSystemDatabases>(context, system_database, "databases", "Contains all the information about all the databases within the current server.");
```
Comment (rschu1ze) [inline src/Storages/System/attachSystemTables.cpp:104]: "all the information" sounds a bit weird. Maybe just "Lists all databases/tab/es/columns of the current server." (l. 104/105/106)

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit c51946a9c923ab0eac503d3368094a18abe0d3dc with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/48350/c51946a9c923ab0eac503d3368094a18abe0d3dc/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/48350/c51946a9c923ab0eac503d3368094a18abe0d3dc/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Storages/System/attachSystemTables.cpp (hunk 9 above)
Comment (nikitamikhaylov) [reply src/Storages/System/attachSystemTables.cpp]: Looks like no.
```
SELECT *
FROM system.licenses
WHERE library_name ILIKE '%rust%'

Query id: 350803e5-b53c-453f-9d7f-1c8f1ccd5d13

Ok.
```

File: src/Storages/System/attachSystemTables.cpp (hunk 2 above)
Comment (nikitamikhaylov) [reply src/Storages/System/attachSystemTables.cpp:142]: I copied it from the documentation.

File: src/Storages/System/attachSystemTables.cpp (hunk 5 above)
Comment (nikitamikhaylov) [reply src/Storages/System/attachSystemTables.cpp:146]: Fixed to align with documentation.

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**